src/ConfigureMentions.php

```diff
@@ -76,19 +86,26 @@ private function configurePostMentions(Configurator $config)
 
         $tag = $config->tags->add($tagName);
 
-        $tag->attributes->add('username');
         $tag->attributes->add('displayname');
         $tag->attributes->add('number')->filterChain->append('#uint');
         $tag->attributes->add('discussionid')->filterChain->append('#uint');
         $tag->attributes->add('id')->filterChain->append('#uint');
 
-        $tag->template = '<a href="{$DISCUSSION_URL}{@discussionid}/{@number}" class="PostMention" data-id="{@id}"><xsl:value-of select="@displayname"/></a>';
+        $tag->template = '
+            <xsl:choose>
+                <xsl:when test="@deleted != 1">
+                    <a href="{$DISCUSSION_URL}{@discussionid}/{@number}" class="PostMention" data-id="{@id}"><xsl:value-of select="@displayname"/></a>
+                </xsl:when>
+                <xsl:otherwise>
+                    <span class="PostMention PostMention--deleted" data-id="{@id}"><xsl:value-of select="@displayname"/></span>
+                </xsl:otherwise>
+            </xsl:choose>';
 
         $tag->filterChain
             ->prepend([static::class, 'addPostId'])
             ->setJS('function(tag) { return flarum.extensions["flarum-mentions"].filterPostMentions(tag); }');
 
-        $config->Preg->match('/\B@(?<username>[a-z0-9_-]+)#(?<id>\d+)/i', $tagName);
+        $config->Preg->match('/\B@"(?<displayname>((?!"#[a-z]{0,3}[0-9]+).)+)"#p(?<id>[0-9]+)\b/', $tagName);
```

Comment:
More repeated regex, but I don't think we can move this into a trait. Might have to keep this one as is

Reply:
It's true that this one is very similar to the user mention regex but it's not identical, I think it's better kept separate anyway.